api/v1beta1/csiaddonsnode_types.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2021.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type CSIAddonsNodeDriver struct {
+	// Name is the name of the CSI driver that this object refers to.
+	// This must be the same name returned by the CSI-Addons GetIdentity()
+	// call for that driver. The name of the driver is in the format:
+	// `rbd/cephfs.csi.ceph.com`
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+
+	// EndPoint is url that contains the ip-address to which the CSI-Addons
+	// side-car listens to.
+	EndPoint string
+}
+
+// CSIAddonsNodeState defines the state of the operation.
+type CSIAddonsNodeState string
+
+const (
+	// Connected represents the Connected state.
+	Connected CSIAddonsNodeState = "Connected"
+
+	// Failed represents the Connection Failed state.
+	Failed CSIAddonsNodeState = "Failed"
+)
+
+// CSIAddonsNodeSpec defines the desired state of CSIAddonsNode
+type CSIAddonsNodeSpec struct {
+	// Driver is the information of the CSI Driver existing on a node.
+	// If the driver is uninstalled, this can become empty.
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	Driver CSIAddonsNodeDriver `json:"driver" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,1,rep,name=drivers"`
```

Comment:
I think part of the `patchStrategy` and `patchMergeKey` options are related to a `drivers` list, but this currently is a single driver instance.

Reply:
do we need to specify the patch strategy,mergekey, etc here?